quinn-proto/src/connection/streams.rs

```diff
@@ -628,7 +653,11 @@ impl Streams {
                 stream.fin_pending = false;
             }
             if stream.is_pending() {
-                self.pending.push_back(id);
+                if priority == stream.priority {
+                    self.pending.push_back(queue_index, id);
+                } else {
+                    queue_index = queue_index.min(self.pending.push(id, stream.priority));
+                }
```

Comment:
If a stream switches from high priority to low priority, would this cause us to skip writing data from streams at intermediate priorities?

Reply:
No, because we take the min of `queue_index` and the result of `self.pending.push`. If the new priority is lower, the new queue index is guaranteed to be larger.